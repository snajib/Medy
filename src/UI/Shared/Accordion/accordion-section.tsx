import React from "react";
import { View, TouchableOpacity, Text, ViewStyle, StyleProp, TextStyle } from "react-native";
import { styles } from "../styles";

interface IAccordianSectionProps {
    active: boolean
    title: string;
    topContainer?: StyleProp<ViewStyle>
    labelText?: StyleProp<TextStyle>
    content?: StyleProp<ViewStyle>
    onClick: (active: boolean) => void
}

interface IAccordianSectionState {
    active: boolean;
}

export class AccordionSection extends React.Component<IAccordianSectionProps, IAccordianSectionState> {
    constructor(props: IAccordianSectionProps) {
        super(props);
        this.state = {
            active: false,
        }
    }

    handleClick = () => () => {
        this.setState({
            active: !this.state.active
        })
    }

    render() {
        return (
            <View style={this.props.topContainer}>
                <TouchableOpacity onPress={this.handleClick()}>
                    <Text style={this.props.labelText}>{this.props.title}</Text>
                </TouchableOpacity>
                <View style={this.state.active ? styles.sectionActive : styles.sectionInActive}>
                    <View>{this.props.children}</View>
                </View>
            </View>
        )
    }
}