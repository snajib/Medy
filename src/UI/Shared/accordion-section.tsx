import React from "react";
import { View, TouchableOpacity, Text, ViewStyle, StyleProp, TextStyle } from "react-native";
import { styles } from "./styles";

interface Props {
    title: string;
    topContainer?: StyleProp<ViewStyle>
    labelText?: StyleProp<TextStyle>
    content?: StyleProp<ViewStyle>
}

interface State {
    active: boolean;
}

export class AccordionSection extends React.Component<Props, State> {
    constructor(props: Props) {
        super(props);
        this.state = {
            active: false,
        }
    }

    handleClick = () => () => {
        this.setState({
            active: !this.state.active
        })
    }

    render() {
        return (
            <View style={this.props.topContainer}>
                <TouchableOpacity onPress={this.handleClick()}>
                    <Text style={this.props.labelText}>{this.props.title}</Text>
                </TouchableOpacity>
                <View style={this.state.active ? styles.sectionActive : styles.sectionInActive}>
                    <View>{this.props.children}</View>
                </View>
            </View>
        )
    }
}
//active ? styles.active : styles.inActive
//<Text>{this.props.children}</Text>