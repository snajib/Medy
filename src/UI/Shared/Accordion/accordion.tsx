import React from "react";
import { Text, View } from "react-native";
import { AccordionSection } from "./accordion-section";
import { styles } from "../styles";

interface Props {
    schema: string[]
}

export class Accordion extends React.Component<Props> {

    public renderSections(list: string[]) {
        return list.map(section => {
            return (
                <AccordionSection
                    labelText={styles.accordionSectionLabel}
                    title={section}>
                    <Text>abc</Text>
                </AccordionSection>
            )
        })
    }

    render() {
        return (
            <View style={styles.topAccordionContainer}>
                {this.renderSections(this.props.schema)}
            </View>
        )
    }
}

/*
            <View style={styles.topAccordionContainer}>
                <AccordionSection
                    labelText={styles.accordionSectionLabel}
                    title="Pictures">
                    <Text>Pics</Text>
                </AccordionSection>
                <AccordionSection
                    labelText={styles.accordionSectionLabel}
                    title="About">
                    <Text>About</Text>
                </AccordionSection>
                <AccordionSection
                    labelText={styles.accordionSectionLabel}
                    title="Technologies">
                    <Text>Tech</Text>
                </AccordionSection>
            </View>


           public renderSections(list: string[]) {
            return list.map(section => {
                <AccordionSection
                    labelText={styles.accordionSectionLabel}
                    title={section}>
                    <Text>abc</Text>
                </AccordionSection>
            })
        }

        render() {
            return (
                <View style={styles.topAccordionContainer}>
                    {this.renderSections(this.props.schema)}
                </View>
            )
        }

        */