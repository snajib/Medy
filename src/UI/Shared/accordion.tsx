import React from "react";
import { Text, View } from "react-native";
import { AccordionSection } from "./accordion-section";
import { styles } from "./styles";

export interface IAccordianProps {
    schema: string[]
}

function accordianSectionClick(active?: boolean): boolean {
    return active = !active;
}

export class Accordion extends React.Component<IAccordianProps> {

    public renderSections(list: string[]) {
        return list.map(section => {
            return (
                <AccordionSection
                    active={false}
                    onClick={accordianSectionClick}
                    labelText={styles.accordionSectionLabel}
                    title={section}>
                    <Text>accordion content</Text>
                </AccordionSection>
            )
        })
    }

    render() {
        return (
            <View style={styles.topAccordionContainer}>
                {this.renderSections(this.props.schema)}
            </View>
        )
    }
}