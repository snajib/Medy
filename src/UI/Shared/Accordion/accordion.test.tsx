import React from 'react';
import renderer from 'react-test-renderer';
import { Accordion } from './accordion';
import { View, Text } from 'react-native';
import { AccordionSection } from './accordion-section';

const mockSchema = ['1', '2', '3', '4', '5'];

it('should render the correct number of sections', () => {
  const accordion = renderer.create(<Accordion schema={mockSchema} />);
  expect(accordion.root.findAllByType(AccordionSection).length).toBe(5);
});

it('should render the correct number of components', () => {
  const accordion = renderer.create(<Accordion schema={mockSchema} />);
  expect(accordion.root.findAllByType(AccordionSection).length).toBe(5);
  expect(accordion.root.findAllByType(Text).length).toBe(10);
  expect(accordion.root.findAllByType(Text)[0].props.children).toEqual(
    mockSchema[0],
  );
  expect(accordion.root.findAllByType(Text)[1].props.children).toEqual(
    'accordion content',
  );
  expect(accordion.root.findAllByType(View).length).toBe(21);
});

it('should render with snapshot', () => {
  const accordion = renderer.create(<Accordion schema={mockSchema} />);
  expect(accordion.toJSON()).toMatchSnapshot();
});
