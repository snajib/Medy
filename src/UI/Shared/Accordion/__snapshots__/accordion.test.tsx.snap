// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with snapshot 1`] = `
<View
  style={
    Object {
      "paddingLeft": "5%",
      "paddingTop": "2.5%",
    }
  }
>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 30,
            "fontStyle": "normal",
            "fontWeight": "bold",
            "paddingTop": "5%",
            "textAlign": "left",
          }
        }
      >
        1
      </Text>
    </View>
    <View
      style={
        Object {
          "height": 0,
          "overflow": "hidden",
        }
      }
    >
      <View>
        <Text>
          accordion content
        </Text>
      </View>
    </View>
  </View>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 30,
            "fontStyle": "normal",
            "fontWeight": "bold",
            "paddingTop": "5%",
            "textAlign": "left",
          }
        }
      >
        2
      </Text>
    </View>
    <View
      style={
        Object {
          "height": 0,
          "overflow": "hidden",
        }
      }
    >
      <View>
        <Text>
          accordion content
        </Text>
      </View>
    </View>
  </View>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 30,
            "fontStyle": "normal",
            "fontWeight": "bold",
            "paddingTop": "5%",
            "textAlign": "left",
          }
        }
      >
        3
      </Text>
    </View>
    <View
      style={
        Object {
          "height": 0,
          "overflow": "hidden",
        }
      }
    >
      <View>
        <Text>
          accordion content
        </Text>
      </View>
    </View>
  </View>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 30,
            "fontStyle": "normal",
            "fontWeight": "bold",
            "paddingTop": "5%",
            "textAlign": "left",
          }
        }
      >
        4
      </Text>
    </View>
    <View
      style={
        Object {
          "height": 0,
          "overflow": "hidden",
        }
      }
    >
      <View>
        <Text>
          accordion content
        </Text>
      </View>
    </View>
  </View>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 30,
            "fontStyle": "normal",
            "fontWeight": "bold",
            "paddingTop": "5%",
            "textAlign": "left",
          }
        }
      >
        5
      </Text>
    </View>
    <View
      style={
        Object {
          "height": 0,
          "overflow": "hidden",
        }
      }
    >
      <View>
        <Text>
          accordion content
        </Text>
      </View>
    </View>
  </View>
</View>
`;
